docs(storage): add guide for manually purging cdn cache (#47348)
supabase/supabase: apps/docs/content/guides/storage/cdn/purge-cdn-cache.mdx

---
id: 'storage-cdn-purge-cache'
title: 'Purge CDN Cache'
description: 'Learn how to manually purge Supabase Storage CDN cache.'
sidebar_label: 'CDN'
---

With Smart CDN enabled, Supabase Storage automatically invalidates the cache when files are updated or deleted. However, there are scenarios where you may need to manually purge the CDN cache for specific objects or entire buckets. The cache purge API allows you to immediately queue cache content invalidation across all CDN edge nodes.

> Manual cache purging is useful when you need to ensure that updates are propagated immediately, or when you want to clear the cache for debugging purposes. Once purged, the next request for that object will be served from the origin server, and the CDN cache will be repopulated.

Reviewer: ```suggestion
Manual cache purging is useful when you need to ensure that updates are propagated as soon as possible, or when you want to clear the cache for debugging purposes. Once purged, the next request for that object will be served from the origin server, and the CDN cache will be repopulated.
```

As later we say "60 seconds", which isn't immediately.

<Admonition type="caution">

> Cache purging requires the **secret key**. Calls made with the anon key or a user JWT will be rejected by the server. Never expose your secret key in client-side code.

Reviewer: ```suggestion
Cache purging requires the **secret key**. The server rejects calls made with the legacy anon key or a user JWT. Never expose your secret key in client-side code.
```

In this case, what is the server? One you're running? Supabase?

</Admonition>

<Admonition type="note">

CDN cache purge is available for [Pro Plan and above](/pricing).

</Admonition>

## Purge a single object

> You can purge the CDN cache for a specific file by providing the exact path to the object. This operation does not support wildcards or recursion—you must specify the complete path of the file you want to invalidate.

Reviewer: ```suggestion
You can purge the CDN cache for a specific file by providing the exact path to the object. This operation does not support wildcards or recursion. You must specify the complete path of the file you want to invalidate.
```

---

> id: 'storage-purge-cdn-cache'

Reviewer 2: all other files under the same folder are using `storage-cdn` as id, not sure if significant for anything
